### ISSUE TITLE
datasette publish cloudrun --memory option

### ISSUE THREAD

**Issue by simonw** [simonw/datasette#694]
Got this error deploying large (603MB) database with Cloud Run
```
X Deploying... Cloud Run error: Container failed to start. Failed to start and then listen on the port defined by the PORT environment variable. Logs for this revi
sion might contain more information.                                                                                                                               
  X Creating Revision... Cloud Run error: Container failed to start. Failed to start and then listen on the port defined by the PORT environment variable. Logs for
   this revision might contain more information.                                                                                                                   
  . Routing traffic...                                                                                                                                             
  ✓ Setting IAM Policy...                                                                                                                                          
Deployment failed                                                                                                                                                  
ERROR: (gcloud.run.deploy) Cloud Run error: Container failed to start. Failed to start and then listen on the port defined by the PORT environment variable. Logs for this revision might contain more information.
```

**Comment by simonw** [comment]
The initial `datasette publish cloudrun` failed, now I can replicate that error by running:
```
gcloud run deploy --allow-unauthenticated --platform=managed \
  --image gcr.io/datasette-222320/datasette non-profit-ethics
```

**Comment by simonw** [comment]
Aha! The logs said "Memory limit of 244M exceeded with 247M used. Consider increasing the memory limit, see https://cloud.google.com/run/docs/configuring/memory-limits"

**Comment by simonw** [comment]
This fixed it (I tried 1Gi first but that gave the same error):
```
gcloud run deploy --allow-unauthenticated --platform=managed \
  --image gcr.io/datasette-222320/datasette non-profit-ethics --memory=2Gi
```

**Comment by simonw** [comment]
I wonder if there's some weird reason that we churn through too much RAM on initial datasette startup here? I wouldn't expect startup to use a huge spike of RAM. Maybe need to profile a bit.

**Comment by simonw** [comment]
Tested that with:
```
datasette publish cloudrun fixtures.db --memory 512Mi --service fixtures-memory-512mi
```
Here's the result in the Google Cloud web console:

<img width="557" alt="fixtures-m…ory-512mi_–_Service_details_–_Cloud_Run_–_datasette_–_Google_Cloud_Platform_-_Mozilla_Firefox" src="https://user-images.githubusercontent.com/9599/76035699-b1c35600-5f07-11ea-8eee-7a1e159e50f7.png">

**Comment by simonw** [comment]
Shipped in [Datasette 0.38](https://datasette.readthedocs.io/en/latest/changelog.html#v0-38).

**Comment by kwladyka** [comment]
@simonw 

Did you find the reason? I had similar situation and I check this on millions ways. I am sure app doesn't consume such memory.

I was trying the app with:
`docker run --rm -it -p 80:80 -m 128M foo`

I was watching app with `docker stats`. Even limited memory by `CMD ["java", "-Xms60M", "-Xmx60M", "-jar", "api.jar"]`.
Checked memory usage by app in code and print bash commands. The app definitely doesn't use this memory. Also doesn't write files.

Only one solution is to change memory to 512M.

It is definitely something wrong with `cloud run`.

I even did special app for testing this. It looks like when I cross very small amount of code / memory / app size in random when, then memory needs grow +hundreds. Nothing make sense here. Especially it works everywhere expect cloud run.

Please let me know if you discovered something more.
